subworkflows/local/prepare_header.nf

```diff
@@ -0,0 +1,39 @@
+// Create a SAM header template from NCBI assembly report and SAMtools .dict
+include { BUILD_SAM_HEADER } from '../../modules/local/build_sam_header'
+
+workflow PREPARE_HEADER {
+
+    take:
+    dict    // file: /path/to/genome.dict
+    report  // file: /path/to/genome.assembly_report.txt
+    source  // file: /path/to/SOURCE (ftp path as string)
+
+    main:
+    ch_versions = Channel.empty()
+
+    // Normalize the dict ID by removing .masked.ncbi if present and join
+    dict_mapped = dict.map { meta, path ->
+        def id = meta.id.replace('.masked.ncbi', '')
+        [id, [meta, path]]
+    }
```

Comment:
I think you don't need that anymore since the sub-workflow is only called on the unmasked file

```suggestion
    // The meta maps differ, so join the channels by meta.id
    dict_mapped = dict.map { meta, path -> [id, [meta, path]] }
```